raiden_contracts/contracts/MonitoringService.sol

```diff
@@ -0,0 +1,321 @@
+pragma solidity ^0.4.23;
+
+import "./Token.sol";
+import "./Utils.sol";
+import "./lib/ECVerify.sol";
+import "./TokenNetwork.sol";
+
+contract MonitoringService is Utils {
+
+    // Token to be used for paying the rewards
+    Token public token; // Only allow RDN?
+
+    // The minimum allowed deposit for a MS
+    uint256 public minimum_deposit;
+
+    // keep track of registered Monitoring Services
+    // A Monitoring Services must deposit to register
+    mapping(address => bool) public registered_monitoring_services;
+
+    // channel_identifier => Struct
+    // Keep track of the rewards per channel
+    mapping(bytes32 => Reward) rewards;
```

Comment:
`bytes32` should be `keccak256(channel_identifier, token_network_address)`. This contract will be used for all token networks and `channel_identifier` is not enough for uniqueness.